### PR TITLE
Refactor realtime signs message contract

### PR DESCRIPTION
**Pull request by PaulJKim**
#### Summary of changes
**Asana Ticket:** [Refactor message contract](https://app.asana.com/0/1201753694073608/1204384030834755/f)

This PR is a refactor that fully eliminates the pattern of tying a `SourceConfig` and message struct together as a tuple. It also takes the interrupting read logic out of the `update_sign()` step in `realtime.ex` and puts it in its own step. This makes things more explicit by avoiding burying the interrupting read logic in the update step.

The changes will cause a lot of tests to fail. To avoid bloating the PR, I will update the tests on a separate branch and PR.

### CODE REVIEW

File: lib/content/audio/predictions.ex [hunk 1]
```diff
@@ -16,12 +16,12 @@ defmodule Content.Audio.Predictions do
   @heavy_rail_routes ["Red", "Orange", "Blue"]
 
   @spec from_sign_content(
-          {Signs.Utilities.SourceConfig.source(), Content.Message.Predictions.t()},
+          Content.Message.Predictions.t(),
```
Comment (PaulJKim) [inline lib/content/audio/predictions.ex:19]: I'm getting a Dialyzer warning here telling me that this typespec is invalid, but it's not obvious to me exactly why. Let me know if you're seeing this as well.

**Review by panentheos — COMMENTED** [review]
Overall this looks great! See the other comment for some long thoughts about that one implementation detail, then I'll do a more thorough review of the whole thing, including the test changes.

File: lib/signs/utilities/audio.ex [hunk 2]
```diff
@@ -17,91 +17,123 @@ defmodule Signs.Utilities.Audio do
...
     false
   end
 
   def should_interrupting_read?(_content, _sign, _line) do
     true
   end
 
+  defp announce_arriving?(source_config, prediction) do
+    source_config
+    |> SourceConfig.get_source_by_stop_and_direction(prediction.stop_id, prediction.direction_id)
```
[the remark this answers is not in the document]
Comment (PaulJKim) [reply lib/signs/utilities/audio.ex:115]: Yeah, I think I agree that we're ultimately not gaining much by delaying the lookup at this point. Apart from the point about potentially removing one of the types though, I think the other factor that I could see in favor of the current approach is that calculating whether or not the sign should interrupting read requires reading `announced_arrivals` from the `Signs.Realtime` struct, at least for one case. We do already pass it down up until one level above where we actually create the `Content.Message.Prediction` but it logically, it seems odd to have that piece of info there. Even if the later lookup isn't completely necessary, I do sort of like how it keeps the textual stuff fully separate from the audio stuff. Does that make sense?

[the remark this answers is not in the document]
Comment (panentheos) [reply lib/signs/utilities/audio.ex:115]: That makes sense, I think this is a good way to leave it.

File: lib/content/audio/predictions.ex (hunk 1 above)
Comment (panentheos) [reply lib/content/audio/predictions.ex:19]: I got this too (in my editor only). I think the language server's instance of Dialyzer gets confused (in a stateful way) sometimes. I was able to "fix" it by making a temporary, valid change to `Message.Predictions` that changed the type (e.g. adding field), which seemed to cause it to re-evaluate and pick up the new code.

**Review by panentheos — APPROVED** [review]
Looks great (after test changes are merged in). This is a nice improvement, thanks! Tested locally.

**Pull request merged**